src/expression.d

```diff
@@ -13976,7 +13985,10 @@ public:
 
     override Expression toBoolean(Scope* sc)
     {
-        e2 = e2.toBoolean(sc);
+        auto ex2 = e2.toBoolean(sc);
```

Comment:
`e2 = e2.toBoolean(sc);` sets `this.e2` to `ErrorExp`. It's not good during error gagging.